script.js

```diff
@@ -1,4 +1,75 @@
+/**
+ * Scissors Paper Stone (Part 1)
+ *
+ * basic version where user inputs one of "scissors", "paper", or "stone", the program internally randomly chooses scissors, paper, or stone, and the program outputs whether the user won, the program won, or it's a draw.
+ * Rules: scissors beats paper, paper beats stone, and stone beats scissors. If both parties choose the same object, it's a draw.
+ */
+
+// Generate a random integer from 1 to max inclusive
+var generateRandomInteger = function (max) {
+  var randomDecimal = Math.random() * max;
+  var randomInteger = Math.floor(randomDecimal) + 1;
+  return randomInteger;
+};
+// Generate random sps item
+var generateRandomSPS = function () {
+  var numSPSoutcome = 3;
+  var randomSPSoutcome = generateRandomInteger(numSPSoutcome);
+  if (randomSPSoutcome == 1) {
+    return "scissors";
+  }
+  if (randomSPSoutcome == 2) {
+    return "paper";
+  }
+  if (randomSPSoutcome == 3) {
+    return "stone";
+  }
+  return "Oops! There may be error!";
+};
+
+var checkIfUserWon = function (userSPS, randomSPS) {
+  if (
+    (userSPS == "scissors" && randomSPS == "paper") ||
+    (userSPS == "paper" && randomSPS == "stone") ||
+    (userSPS == "stone" && randomSPS == "scissors") ||
+    //include reverse game
+    (userSPS == "reversed scissors" && randomSPS == "stone") ||
+    (userSPS == "reversed paper" && randomSPS == "scissors") ||
+    (userSPS == "reversed stone" && randomSPS == "paper")
```

Comment:
Scissors, paper, stone (and their coresponding reverses) can be stored in variables so that you won't have to input "stone","scissors", etc. storing them in variables can also minimise potential spelling errors.